lib/before.js

```diff
@@ -99,8 +99,10 @@ module.exports = function(scope, cb) {
 	// Determine default values based on the
 	// available scope.
 
+	// Check if slash delimiter exists.
+	var slashIndex = scope.id.lastIndexOf('/');
 	_.defaults(scope, {
-		entity: _.str.capitalize(scope.id) + 'Controller',
+		entity: (slashIndex === -1 ? _.str.capitalize(scope.id) : _.str.splice(scope.id, slashIndex + 1, 1, scope.id.charAt(slashIndex + 1).toUpperCase())) + 'Controller',
```

Comment:
@1001v In case this helps, here's how I'd approach it these days:


```
var path = require('path');//<< at the top w /the other requires

// ...

// Determine the "entity" for this controller.
// This expects strings like `user` or `pet`, but it also tolerates `foo/bar/pet`
// > (in which case "pet" is the entity)
var entity = path.basename(scope.id);
entity = _.capitalize(entity);
entity += 'Controller';

// ...
entity: entity.